File: LeetcodeJava/src/main/java/org/leetcode/algorithm/P3110.java
package org.leetcode.algorithm;

/**
 * Author: Deean
 * Date: 2025-03-09 10:54
 * FileName: src/main/java/org/leetcode/algorithm
 * Description:
 */

public class P3110 {
    class Solution {
        public int scoreOfString(String s) {
            char[] ss = s.toCharArray();
            int score = 0;
            for (int i = 0; i < ss.length - 1; i++) {
                score += Math.abs(ss[i] - ss[i + 1]);
            }
            return score;
        }
    }

    public static void main(String[] args) {
        Solution s = new P3110().new Solution();
        Object ans = s.scoreOfString("hello");
        System.out.println(ans);
    }
}

File: LeetcodeCC/algorithm/P3110.cpp
/**
 * Author: Deean
 * Date: 2025-05-11 22:15
 * FileName: P3110.cpp
 * Description: 3110. 字符串的分数
 */

#include "../lib/codec.h"

using namespace std;

class Solution {
public:
    int scoreOfString(string s) {
        int score = 0;
        for (int i = 1; i < s.length(); i++) {
            score += abs(s[i] - s[i - 1]);
        }
        return score;
    }
};

int main() {
    int ans = Solution().scoreOfString("hello");
    cout << ans << endl;
    return 0;
}
